gcpkms/keysource.go

```diff
@@ -203,8 +207,8 @@ func (key *MasterKey) TypeToIdentifier() string {
 	return KeyTypeIdentifier
 }
 
-// newKMSClient returns a GCP KMS client configured with the credentialJSON
-// and/or grpcConn, falling back to environmental defaults.
+// newKMSClient returns a GCP KMS client configured with the credentialJSON,
+// tokenSource and/or grpcConn, falling back to environmental defaults.
```

Comment:
I don't think the GCP authentication method introduced in this PR is quite what you would call "token source". A token source method would be similar to the equivalents of AWS and Azure:


https://github.com/getsops/sops/blob/5355c249aa8ea413bea6b2ea20a029cb21048e99/kms/keysource.go#L72-L76

https://github.com/getsops/sops/blob/5355c249aa8ea413bea6b2ea20a029cb21048e99/azkv/keysource.go#L58-L62